sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/TimeSeriesInsightsClient.cs

```diff
@@ -239,5 +247,777 @@ public virtual Response<TimeSeriesModelSettings> UpdateModelSettingsDefaultTypeI
                 throw;
             }
         }
+
+        /// <summary>
+        /// Gets Time Series instances asynchronously from the environment in pages.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The pageable list <see cref="AsyncPageable{TimeSeriesInstance}"/> of Time Series instances belonging to the TSI environment and the http response.</returns>
+        /// <remarks>
+        /// For more samples, see <see href="https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/samples">our repo samples</see>.
+        /// </remarks>
+        /// <example>
+        /// <code snippet="Snippet:TimeSeriesInsightsSampleGetInstances">
+        /// </code>
+        /// </example>
+        public virtual AsyncPageable<TimeSeriesInstance> GetInstancesAsync(CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
+            scope.Start();
+
+            try
+            {
+                async Task<Page<TimeSeriesInstance>> FirstPageFunc(int? pageSizeHint)
+                {
+                    using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TimeSeriesInsightsClient)}.{nameof(GetInstances)}");
```

Comment:
why is it twice? Also at line 265?

Reply:
I actually noticed the same thing in ADT client code, where a scope is created at the start of the method, and at the start of the FirstPageFunc and NextPageFunc.

@azabbasi is that necessary?